hepdata_validator/data_file_validator.py

```diff
@@ -137,6 +156,84 @@ def validate(self, **kwargs):
         else:
             return True
 
+    def check_independent_variable_values(self, file_path, data_item):
+        """
+        Check that 'independent_variables' values are not a range like 1.7-4.7.
+
+        :param data_item: YAML document from submission.yaml
+        :return: raise ValidationError if not numeric
+        """
+        if 'independent_variables' in data_item:
+            for i, var in enumerate(data_item['independent_variables']):
+                if 'values' in var:
+                    for j, v in enumerate(var['values']):
+                        if 'value' in v and isinstance(v['value'], str) and '-' in v['value']:
+                            m = re.match(r'^[+-]\d+\.?\d?([eE][+-]\d+)?-\d+\.?\d?([eE][+-]\d+)?$', v['value'])
+                            if not m:
```

Comment:
I think there are some flaws here with both the logic and the regex.  The error should be raised if the pattern is matched, i.e. `if m`.  It looks like your tests pass because none of your test cases actually match the regex, which currently requires either `+` or `-` before the first number, no `+` or `-` before the second number, and no spaces either side of the hyphen in between.  I caught this when testing a submission with strings like `value: 'Mono-V overlap removal'` which should pass validation because there are no numbers, but currently fail validation.  I wonder if a simpler approach would be to match a regex like `r'^(.+)\s?-\s?(.+)$'` and check that the strings in brackets can be cast to a `float` without a `TypeError`.  Otherwise, I think the existing approach would work if you change `if not m` to `if m` and modify the regex so that the first `[+-]` is optional, insert `[+-]?` before the second number, and allow optional spaces `\s?` either side of the separating `-`.

Reply:
Apologies for the dodgy logic and regex - not sure what I was thinking! (Probably distracted by trying to sort out the error messages and multiple errors per file at the same time.) Unfortunately splitting on `-` (whether by regex or `split`) won't work easily because of the possibility of negative numbers and exponents, so I had to use a regex for the numbers. I had a more thorough regex in the schema at one point so I'll go back and check against that as well as against your suggestions.